tests/e2e/kubetest2-kops/aws/s3.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"k8s.io/klog/v2"
+)
+
+// We need to pick some region to query the AWS APIs through, even if we are not running on AWS.
+const defaultRegion = "us-east-2"
+
+// It contains S3Client, an Amazon S3 service client that is used to perform bucket
+// and object actions.
+type awsClient struct {
+	S3Client *s3.Client
+}
+
+func NewAWSClient(ctx context.Context) (*awsClient, error) {
+	cfg, err := awsconfig.LoadDefaultConfig(ctx,
+		awsconfig.WithRegion(defaultRegion))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load AWS config: %w", err)
+	}
+	return &awsClient{
+		S3Client: s3.NewFromConfig(cfg),
+	}, nil
+}
+
+// AWSBucketName constructs an unique bucket name using the AWS account ID on region us-east-2
+func AWSBucketName(ctx context.Context) (string, error) {
+	config, err := awsconfig.LoadDefaultConfig(ctx, awsconfig.WithRegion(string(types.BucketLocationConstraintUsEast2)))
+	if err != nil {
+		return "", fmt.Errorf("failed to load AWS config: %w", err)
+	}
+
+	stsSvc := sts.NewFromConfig(config)
+	callerIdentity, err := stsSvc.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return "", fmt.Errorf("building AWS STS presigned request: %w", err)
+	}
+
+	// Add timestamp suffix
+	timestamp := time.Now().Format("01022006")
+	bucket := fmt.Sprintf("k8s-infra-kops-%s", *callerIdentity.Account)
+	bucket = fmt.Sprintf("%s-%s", bucket, timestamp)
```

Comment:
Why don't we use a random name here? Isn't there a risk that if cleanup fails, the account gets reused, we are either stuck with an existing bucket that's not cleaned up or not being able to create another bucket for that day?

Reply:
We could even make it more unique by having `timestamp := time.Now().Format("20060102150405")`  but I'm not opposed to use `math/rand`